src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -438,12 +438,12 @@ abstract class ManagedStrategy(
         if (inIgnoredSection(iThread)) return false
         newSwitchPoint(iThread, codeLocation, tracePoint)
         // Try to acquire the monitor
-        if (!monitorTracker.acquireMonitor(iThread, monitor)) {
-            failIfObstructionFreedomIsRequired { "Obstruction-freedom is required but a lock has been found" }
+        while (!monitorTracker.acquireMonitor(iThread, monitor)) {
+            failIfObstructionFreedomIsRequired {
+                "Obstruction-freedom is required but a lock has been found"
```

Comment:
Please enhance the message and extract it to a constant

Reply:
I extracted the messages into constants. 

How exactly do you wish to improve the error messages themself?

Reply:
Just make it simpler to read, something like "The algorithm should be non-blocking, but an active lock is detected"

Reply:
Note that we do not check for other progress guarantees than obstruction-freedom, so I would suggest using "non-blocking" instead of "obstruction-freedom" everywhere for simplicity.

Reply:
DOne